commit 86081fa78ce2e3346db77fb55df93c550cf3f57a
Author: Jonathan Ellis <jbellis@apache.org>
Date:   Wed Jun 19 13:55:40 2013 -0500

    make TokenMetadata.relocatingTokens a ConcurrentMap
    patch by jbellis; reviewed by brandonwilliams for CASSANDRA-5634

diff --git a/src/java/org/apache/cassandra/locator/TokenMetadata.java b/src/java/org/apache/cassandra/locator/TokenMetadata.java
index 925a811..1f6682b 100644
--- a/src/java/org/apache/cassandra/locator/TokenMetadata.java
+++ b/src/java/org/apache/cassandra/locator/TokenMetadata.java
@@ -85,7 +85,7 @@ public class TokenMetadata
     private final Set<Pair<Token, InetAddress>> movingEndpoints = new HashSet<Pair<Token, InetAddress>>();
 
     // tokens which are migrating to new endpoints
-    private final Map<Token, InetAddress> relocatingTokens = new HashMap<Token, InetAddress>();
+    private final ConcurrentMap<Token, InetAddress> relocatingTokens = new ConcurrentHashMap<Token, InetAddress>();
 
     /* Use this lock for manipulating the token map */
     private final ReadWriteLock lock = new ReentrantReadWriteLock(true);
